src/ensembl/src/shared/tooltip/tooltip-helper.ts

```diff
@@ -15,62 +15,67 @@ const leftSide = [Position.LEFT_TOP, Position.LEFT_BOTTOM];
 const rightSide = [Position.RIGHT_TOP, Position.RIGHT_BOTTOM];
 
 type FindOptimalPositionParams = {
-  intersectionEntry: IntersectionObserverEntry;
+  tooltipBoundingRect: ClientRect;
+  rootBoundingRect: ClientRect;
   anchorBoundingRect: ClientRect;
   position: Position;
 };
 
 export const findOptimalPosition = (params: FindOptimalPositionParams) => {
-  const {
-    intersectionEntry: { isIntersecting }
-  } = params;
-  let { position } = params;
-
-  if (isIntersecting) {
-    return position;
-  }
-
-  return adjustPosition(params);
-};
-
-const adjustPosition = (params: FindOptimalPositionParams) => {
   const possiblePositions = getPossiblePositions(params);
-  for (let position of possiblePositions) {
-    if (willStayWithinBounds({ ...params, position })) {
-      return position;
-    }
-  }
 
-  // if no better position was found, return original position
-  return params.position;
+  return possiblePositions.reduce(
+    (result, position) => {
+      const outOfBoundsArea = getTooltipOutOfBoundsArea({
+        ...params,
+        position
+      });
+      if (outOfBoundsArea < result.outOfBoundsArea) {
+        return {
+          position,
+          outOfBoundsArea
+        };
+      } else {
+        return result;
+      }
+    },
+    { position: params.position, outOfBoundsArea: Infinity }
+  ).position;
 };
 
 const getPossiblePositions = (params: FindOptimalPositionParams) => {
   const { position } = params;
   if (topRow.includes(position)) {
-    return [...topRow, ...bottomRow];
+    return [...preferredFirst(topRow, position), ...bottomRow];
   } else if (bottomRow.includes(position)) {
-    return [...bottomRow, ...topRow];
+    return [...preferredFirst(bottomRow, position), ...topRow];
   } else if (leftSide.includes(position)) {
-    return [...leftSide, ...rightSide];
+    return [...preferredFirst(leftSide, position), ...rightSide];
   } else if (rightSide.includes(position)) {
-    return [...rightSide, ...leftSide];
+    return [...preferredFirst(rightSide, position), ...leftSide];
   } else {
     return [];
   }
 };
 
-const willStayWithinBounds = (params: FindOptimalPositionParams) => {
+const preferredFirst = (positions: Position[], preferredPosition: Position) => {
```

Comment:
Is it possible to think of a better name for this function?

Reply:
Renamed the function to `bringToFront` and moved it in its own file in src/shared/utils